expr/func_absolute.go

```diff
@@ -31,16 +31,19 @@ func (s *FuncAbsolute) Exec(dataMap DataMap) ([]models.Series, error) {
 		return nil, err
 	}
 
-	for i, serie := range series {
-		series[i].Target = fmt.Sprintf("absolute(%s)", serie.Target)
-		series[i].Tags = serie.CopyTagsWith("absolute", "1")
-		series[i].QueryPatt = fmt.Sprintf("absolute(%s)", serie.QueryPatt)
-		series[i].Datapoints = pointSlicePool.Get().([]schema.Point)
+	outSeries := make([]models.Series, 0, len(series))
```

Comment:
I see your argument for consistency, but it does seem a bit silly to reallocate output slices when for each incoming series there is 1 outgoing. we could reuse the input slice

Reply:
There is an existing code path that allows for the exact same slice to be passed to multiple target branches. That's exactly the issue in #829 

Now, that *could* be fixed as well, but the copy has to happen somewhere and we decided CoW a long time ago.

Reply:
ah yes. can you also update devdocs/expr.md to clarify that COW now applies to the full series structures being passed into functions.